jwe/keycrypter.go

```diff
@@ -0,0 +1,53 @@
+package jwe
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1"
+	"errors"
+)
+
+var (
+	ErrUnsupportedKeyType      = errors.New("unsupported key type")
+	ErrUnsupportedKeyAlgorithm = errors.New("unsupported key algorithm")
+)
+
+type KeyAlgorithm string
+
+var RSA_OAEP = KeyAlgorithm("RSA-OAEP")
+
+func rsaEncrypt(key *rsa.PublicKey, cek []byte, alg KeyAlgorithm) ([]byte, error) {
```

Comment:
any particular reason why you are using functions here instead of a struct with methods? Just wondering. The latter might be better if we want to make an interface out of it.

Reply:
Changed to struct with Encrypt and Decrypt. For the future